src/en/mangaowlto/src/eu/kanade/tachiyomi/extension/en/mangaowlto/MangaOwlTo.kt

```diff
@@ -0,0 +1,183 @@
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import uy.kohesive.injekt.injectLazy
+import java.util.concurrent.TimeUnit
+
+class MangaOwlTo(
+    private val collection: String = "manga",
+    extraName: String = "",
+    private val genresList: List<Genre> = listOf(),
+) : ConfigurableSource, HttpSource() {
+    override val name: String = "MangaOwl.To $extraName"
+    override val lang = "en"
+    override val supportsLatest = true
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    private val defaultDomain: String = preferences.getString(MIRROR_PREF_KEY, MIRROR_PREF_DEFAULT_VALUE) ?: MIRROR_PREF_ENTRY_VALUES[0]
+    override val baseUrl = "https://api.$defaultDomain/v1"
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        val mirrorPref = ListPreference(screen.context).apply {
+            key = MIRROR_PREF_KEY
+            title = MIRROR_PREF_TITLE
+            entries = MIRROR_PREF_ENTRIES
+            entryValues = MIRROR_PREF_ENTRY_VALUES
+            setDefaultValue(MIRROR_PREF_DEFAULT_VALUE)
+            summary = "%s"
+
+            setOnPreferenceChangeListener { _, newValue ->
+                val selected = newValue as String
+                val index = findIndexOfValue(selected)
+                val entry = entryValues[index] as String
+                preferences.edit().putString(MIRROR_PREF_KEY, entry).commit()
+            }
+        }
+        screen.addPreference(mirrorPref)
+    }
+
+    private val json: Json by injectLazy()
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .connectTimeout(1, TimeUnit.MINUTES)
+        .readTimeout(1, TimeUnit.MINUTES)
+        .writeTimeout(1, TimeUnit.MINUTES)
+        .build()
+
+    override fun popularMangaRequest(page: Int) =
+        GET("$baseUrl/stories?type=$collection&ordering=-view_count&page=$page", headers)
+
+    override fun popularMangaParse(response: Response) =
+        json.decodeFromString<MangaOwlToStories>(response.body.string()).toMangasPage()
+
+    // Latest
+
+    override fun latestUpdatesRequest(page: Int) =
+        GET("$baseUrl/stories?type=$collection&ordering=-modified_at&page=$page", headers)
+
+    override fun latestUpdatesParse(response: Response) = popularMangaParse(response)
+
+    // Search
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return if (query.isNotEmpty() || filters.isEmpty()) {
+            // Search won't work together with filter
+            GET("$baseUrl/search?q=$query&page=$page", headers)
```

Comment:
Use `baseUrl.toHttpUrl().newBuilder()` then add the query to encode it properly.

Reply:
I understand what you mean, but isn't using `.toHttpUrl()` sufficient?

Also, there are like 650 occurrences in other extensions also doing this kind of `GET("...", header)` without even using `.toHttpUrl()` at all.  Shouldn't we fix them all if it's a potential issue?

Reply:
Calling `toHttpUrl()` alone doesn't do anything. If the `query` contains characters like `&` (query parameter separater) or `+` (encoded whitespace), concatenating the string will produce a totally wrong URL.

We are aware of these issues and will eventually add lint rules to enforce encoding query text.